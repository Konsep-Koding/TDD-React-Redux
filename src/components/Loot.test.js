import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Loot }  from './Loot'

configure({ adapter: new Adapter(), disableLifecycleMethods: false });

describe('Loot Component', () => {
    const mockFetchbitcoin = jest.fn() 
    let props = {balance: 10, bitcoin: {} }
    let loot = shallow(<Loot {...props} fetchBitcoin={mockFetchbitcoin} />)


     it('should  render properly', () => {
        expect(loot).toMatchSnapshot();
     });
     describe('when mounted', () => {
    const mockFetchbitcoin = jest.fn() 

         beforeEach(() => {
             props.fetchBitcoin = mockFetchbitcoin
             loot = mount(<Loot {...props}  fetchBitcoin={mockFetchbitcoin} />)
         });

         it('should dispatches the fetchBitcoin() method receives from props ', () => {
             expect(mockFetchbitcoin).toHaveBeenCalled()
         });
     });

     describe('when there are valid bitcoin props', () => {
        beforeEach(() => {
          props.fetchBitcoin = mockFetchbitcoin
          props = { balance: 10, bitcoin: { bpi: { USD: { rate: '1,000' } } } };
          loot = shallow(<Loot {...props} fetchBitcoin={mockFetchbitcoin}/>);
        });
    
        it('displays the correct bitcoin value', () => {
          expect(loot.find('h3').text()).toEqual('Bitcoin balance: 0.01');
        });
      });
});
