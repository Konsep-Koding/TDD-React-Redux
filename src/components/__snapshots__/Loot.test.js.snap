// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loot Render 1`] = `ShallowWrapper {}`;

exports[`Loot renders properly 1`] = `ShallowWrapper {}`;
